frontend/src/metabase/visualizations/components/TableSimple.jsx

```diff
@@ -108,9 +108,17 @@ export default class TableSimple extends Component {
     if (sortColumn != null) {
       rowIndexes = _.sortBy(rowIndexes, rowIndex => {
         let value = rows[rowIndex][sortColumn];
+        const col = cols[sortColumn];
         // for strings we should be case insensitive
         if (typeof value === "string") {
-          value = value.toLowerCase();
+          if (isID(col)) {
+            value = parseInt(value, 10);
+          } else {
+            value = value.toLowerCase();
+          }
```

Comment:
I suppose it won't work as expected for UUID or any other non-numeric PK. I've tested locally with [the sample MongoDB](https://github.com/metabase/metabase-qa#mongo-4)

1. If a UUID starts with numeric characters, `parseInt` will just extract them and drop the rest which works, but probably doesn't make a lot of sense.
2. If a UUID starts with letters (e.g. `ff2ae2f1-ab0d-45b4-8f8b-604df3ca92e4`), `parseInt` will return `NaN` and we'll probably enter some weird behaviour zone

Should we ensure that when we do `parseInt`, we're certain it is actually a numeric value we're dealing with?

**Example of scenario 1**

`originalValue` is the cell value, `value` is what will be used after `parseInt` and other checks

![CleanShot 2022-04-21 at 13 49 26@2x](https://user-images.githubusercontent.com/17258145/164461929-f2ad7473-043b-495a-874b-b05344d37366.png)

Reply:
I've added an additional check to make sure the string is a positive integer: https://github.com/metabase/metabase/pull/21838/commits/70be543e54927a3b363d3aa78ba7783c0cc60140

![Screen Shot 2022-04-21 at 10 44 34 AM](https://user-images.githubusercontent.com/13057258/164522949-0bcff7e8-2c7c-4e58-8c47-55dd3dc353f0.png)